README.rst

```diff
@@ -48,12 +57,212 @@ Install
 Usage
 =====
 
-1. Create configurations steps based on ``BaseConfigurationStep`` class.
+Key Concepts
+------------
+
+- **Configuration Model**: A `Pydantic <https://docs.pydantic.dev/>`_ model defining the structure and validation rules for your configuration.
+- **Configuration Step**: A class that implements the actual configuration logic using the validated configuration model.
+
+Getting Started
+---------------
+
+Define a Configuration Model
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    from pydantic import Field
+    from django_setup_configuration import ConfigurationModel, DjangoModelRef
+
+    class UserConfigurationModel(ConfigurationModel):
+        # Use Pydantic's validation features
+        add_to_groups: list[str] = Field(
+            default_factory=list,
+            description="Groups to add the user to"
+        )
+
+        # Reference Django model fields with automatic type inference
+        username = DjangoModelRef("auth.User", "username", default="admin")
+
+        # You can optionally override the inferred type (overriding the type is
+        # required for fields that can not be unambiguously mapped to a Python type,
+        # such as relational fields or third-party fields).
+        is_staff: Optional[int] = DjangoModelRef("auth.User", "username", default=0)
```

Comment:
Can we replace Optional with the more idiomatic `| None` syntax? Or do we need to support ancient python versions?